### ISSUE TITLE
Virtual environement?

### ISSUE THREAD

**Issue by grischard** [esphome/esphome-vscode#13]
Has anyone managed to use this with a python venv?

**Comment by foxsam21** [comment]
I am also looking for this. The issue I have is that I cannot flash from vcs because esphome is not found. I imagine that there should be a way to activate the venv for this plugin. Besides for that issue it seems everything works fine.

**Comment by glmnet** [comment]
Not that I tried but what if you start code from the venv?

Just in case make sure you close all code instances. 

Then from the venv enter `code .`

**Comment by foxsam21** [comment]
:) Thanks. That worked. I was thinking of much more complicated ways to do it... This was simple and to the point.

**Comment by glmnet** [comment]
Did this work for you too @grischard? can we close this then?

**Comment by grischard** [comment]
Yes, that works for me, but would fit nicely in the documentation :)

**Comment by foxsam21** [comment]
Looking in the task provider files in my extension I found that I can add the path to esphome and then it works without having to start it from within the venv.
`}, 'esphome compile', 'ESPHome', new vscode.ShellExecution('<path to venv>/bin/esphome ${relativeFile} run --upload-port OTA'));`
this was in .vscode/extensions/esphome.esphome-vscode-0.3.0/out/EsphomeTaskProvider.js line 7 for me.

Maybe it would be possible to add a way to configure this in a configuration file so someone can have an alternate esphome binary.
I didn't try this but adding esphome to the system path should also work.

**Comment by glmnet** [comment]
> but adding esphome to the system path should also work.

you're right
